pkg/git/gitclient/gitclient.go

```diff
@@ -250,6 +252,17 @@ func (g *GitClient) SetTokenAuth(token string, username string) {
 	g.Opts.Auth = &http.BasicAuth{Password: token, Username: username}
 }
 
+func (g *GitClient) Validate(ctx context.Context) error {
```

Comment:
```suggestion
func (g *GitClient) ValidateRemoteExists(ctx context.Context) error {
```

Reply:
validate might be too broad, we only want to check if the remote exists in this case